qiskit/circuit/gate.py

```diff
@@ -71,3 +71,50 @@ def label(self, name):
             self._label = name
         else:
             raise TypeError('label expects a string or None')
+
+    @staticmethod
+    def _argument_expansion_1(qarg):
+        for arg0 in qarg:
+            yield [arg0], []
+
+    @staticmethod
+    def _argument_expansion_2(qarg0, qarg1):
+        if len(qarg0) == len(qarg1):
+            for arg0, arg1 in zip(qarg0, qarg1):
+                yield [arg0, arg1], []
+        elif len(qarg0) == 1:
+            for arg1 in qarg1:
+                yield [qarg0[0], arg1], []
+        elif len(qarg1) == 1:
+            for arg0 in qarg0:
+                yield [arg0, qarg1[0]], []
+        else:
+            raise QiskitError('Not sure how to combine these two qubit arguments:\n %s\n %s' %
+                              (qarg0, qarg1))
+
+    @staticmethod
+    def _argument_expansion_3(qarg0, qarg1, qarg2):
+        if len(qarg0) == len(qarg1) == len(qarg2):
+            for arg0, arg1, arg2 in zip(qarg0, qarg1, qarg2):
+                yield [arg0, arg1, arg2], []
+        else:
+            raise QiskitError(
+                'Not sure how to combine these three qubit arguments:\n %s\n %s\n %s' %
+                (qarg0, qarg1, qarg2))
+
+    def argument_expansion(self, qargs, cargs):
+        if len(qargs) != self.num_qubits or cargs:
```

Comment:
Can we add documentation around the standard broadcasting rules we support for Gates?

Reply:
Added in 5276755

Reply:
Added in 5276755